src/connectors/youtube.js

```diff
@@ -177,14 +177,14 @@ function getVideoCategory(videoId) {
 	if (videoId === null) {
 		return null;
 	}
-	if (categoryCache.has(videoId)) {
+	if (!categoryCache.has(videoId)) {
 		fetchCategoryId(videoId).then((category) => {
 			if (category === null) {
-				throw Error(`Failed to resolve category for ${videoId}`);
+				console.log(`Failed to resolve category for ${videoId}`);
 			}
 			categoryCache.set(videoId, category);
-		}).catch((err) => {
-			console.log(err);
+		}).catch((error) => {
```

Comment:
I noticed this `catch` is unused, because `fetchCategoryId` never throws any error.